File: backend/internal/dto/exam.go
package dto

type ExamTemplateResp struct {
	ID            uint   `json:"id"`
	SubjectID     uint   `json:"subject_id"`
	Name          string `json:"name"`
	Duration      int    `json:"duration"`
	TotalScore    int    `json:"total_score"`
	Year          int    `json:"year"`
	QuestionCount int    `json:"question_count"`
}

type StartExamResp struct {
	RecordID     uint            `json:"record_id"`
	TemplateID   uint            `json:"template_id"`
	TemplateName string          `json:"template_name"`
	Duration     int             `json:"duration"`
	TotalScore   int             `json:"total_score"`
	Questions    []ExamQuesResp  `json:"questions"`
	Answers      map[uint]string `json:"answers"`
	Elapsed      int             `json:"elapsed"`
	Status       string          `json:"status"`
}

type ExamQuesResp struct {
	ID         uint   `json:"id"`
	Type       string `json:"type"`
	Content    string `json:"content"`
	Options    string `json:"options"`
	Difficulty string `json:"difficulty"`
	Score      int    `json:"score"`
	SortOrder  int    `json:"sort_order"`
}

type SubmitAnswerReq struct {
	QuestionID uint   `json:"question_id" binding:"required"`
	Answer     string `json:"answer"`
}

type ExamResultResp struct {
	RecordID        uint                  `json:"record_id"`
	TemplateName    string                `json:"template_name"`
	SubjectID       uint                  `json:"subject_id"`
	Score           int                   `json:"score"`
	TotalScore      int                   `json:"total_score"`
	Duration        int                   `json:"duration"`
	CorrectCount    int                   `json:"correct_count"`
	TotalCount      int                   `json:"total_count"`
	Status          string                `json:"status"`
	StartedAt       string                `json:"started_at"`
	FinishedAt      string                `json:"finished_at"`
	SectionAccuracy []SectionAccuracyResp `json:"section_accuracy"`
	AnswerDetails   []AnswerDetailResp    `json:"answer_details"`
}

type SectionAccuracyResp struct {
	Type         string  `json:"type"`
	TotalCount   int     `json:"total_count"`
	CorrectCount int     `json:"correct_count"`
	Accuracy     float64 `json:"accuracy"`
}

type AnswerDetailResp struct {
	QuestionID uint   `json:"question_id"`
	Type       string `json:"type"`
	Content    string `json:"content"`
	Options    string `json:"options"`
	YourAnswer string `json:"your_answer"`
	CorrectAns string `json:"correct_answer"`
	IsCorrect  int    `json:"is_correct"`
	Score      int    `json:"score"`
	Analysis   string `json:"analysis"`
}

type ExamRecordResp struct {
	ID           uint   `json:"id"`
	TemplateID   uint   `json:"template_id"`
	TemplateName string `json:"template_name"`
	SubjectID    uint   `json:"subject_id"`
	Score        int    `json:"score"`
	TotalScore   int    `json:"total_score"`
	Duration     int    `json:"duration"`
	Status       string `json:"status"`
	StartedAt    string `json:"started_at"`
	FinishedAt   string `json:"finished_at"`
	CreatedAt    string `json:"created_at"`
}

File: frontend/src/types/exam.ts
export interface ExamTemplateResp {
  id: number
  subject_id: number
  name: string
  duration: number
  total_score: number
  year: number
  question_type: string
  question_count: number
}

export interface ExamQuesResp {
  id: number
  type: string
  content: string
  case_material?: string
  options: string
  blank_options?: string
  difficulty: string
  score: number
  sort_order: number
}

export interface StartExamResp {
  record_id: number
  template_id: number
  template_name: string
  duration: number
  total_score: number
  questions: ExamQuesResp[]
  answers: Record<number, string>
  elapsed: number
  status: string
}

export interface SubmitAnswerReq {
  question_id: number
  answer: string
}

export interface AnswerDetailResp {
  question_id: number
  exam_answer_id: number
  type: string
  content: string
  options: string
  blank_options?: string
  your_answer: string
  correct_answer: string
  is_correct: number
  score: number
  analysis: string
}

export interface SectionAccuracyResp {
  type: string
  total_count: number
  correct_count: number
  accuracy: number
}

export interface ExamResultResp {
  record_id: number
  template_name: string
  subject_id: number
  score: number
  total_score: number
  duration: number      // 秒
  correct_count: number
  total_count: number
  accuracy: number      // 本次考试整体正确率 %
  status: string
  started_at: string
  finished_at: string
  section_accuracy: SectionAccuracyResp[]
  answer_details: AnswerDetailResp[]
}

export interface ExamRecordResp {
  id: number
  template_id: number
  template_name: string
  subject_id: number
  score: number
  total_score: number
  duration: number
  correct_count: number
  accuracy: number
  status: string
  started_at: string
  finished_at: string
  created_at: string
}
